pkg/streamable/streamable.go

```diff
@@ -201,27 +152,65 @@ func unmarshalField(bytes []byte, fieldType reflect.Type, fieldValue reflect.Val
 		}
 		newInt := util.BytesToUint64(newVal)
 		fieldValue.SetUint(newInt)
+	case reflect.Array:
+		switch fieldValue.Type().Elem().Kind() {
```

Comment:
It might be nice to add a default case here and return an error for an unimplemented type, similar to what we do when Marshalling